Library/Formula/kibana.rb

```diff
@@ -13,36 +12,58 @@ class Kibana < Formula
     sha256 "5d102f006b25dae70bd62c99d9787abe10111460bc2119e5e5ceb412d6335dae" => :mountain_lion
   end
 
-  depends_on "node"
+  resource "node" do
+    url "https://nodejs.org/dist/v0.12.7/node-v0.12.7.tar.gz"
+    sha256 "b23d64df051c9c969b0c583f802d5d71de342e53067127a5061415be7e12f39d"
+  end
 
   def install
-    ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+    resource("node").stage buildpath/"node"
+    cd buildpath/"node" do
+      system "./configure", "--prefix=#{libexec}/node"
+      system "make", "install"
+    end
 
-    system "npm", "install"
-    system "npm", "install", "grunt-cli"
-    system "npm", "install", "bower"
-    system "./node_modules/.bin/bower", "install"
-    system "./node_modules/.bin/grunt", "build", "--force"
+    # do not download binary installs of Node.js
+    inreplace buildpath/"tasks/build/index.js" do |s|
+      s.gsub!(%r{('_build:downloadNodeBuilds:\w+',)}, "// \\1")
+    end
 
-    dist_dir = buildpath/"build/dist/kibana"
+    # do not build packages for other platforms
+    inreplace buildpath/"tasks/config/platforms.js" do |s|
+      s.gsub!(%r{('(linux-x64|linux-x86|windows)',?(?!;))}, "// \\1")
+    end
 
-    rm_f dist_dir/"bin/*.bat"
+    # do not build zip package
+    inreplace buildpath/"tasks/build/archives.js" do |s|
+      s.gsub!(%r{(await exec\('zip'.*)}, "// \\1")
+    end
 
-    prefix.install dist_dir/"src"
-    (etc/"kibana").mkpath
-    (var/"lib/kibana/plugins").mkpath
+    ENV.prepend_path "PATH", prefix/"libexec/node/bin"
+    system "npm", "install"
+    system "npm", "run", "build"
+    mkdir "tar"
+    system "tar", "--strip-components", "1", "-xf", Dir["target/kibana-*-darwin-x64.tar.gz"].first, "-C", "tar"
```

Comment:
Maybe work using `Dir["target/kibana-*.tar.gz"]` as the glob just to avoiding definitely breaking 32-bit or Linuxbrew builds.